application/lib/view_model/home_screen_view_model.dart

```diff
@@ -0,0 +1,57 @@
+import 'package:application/model/product_dto.dart';
+import 'package:application/model/product_model.dart';
+import 'package:application/model/user_dto.dart';
+import 'package:application/model/user_model.dart';
+import 'package:application/service/home_screen_service.dart';
+import 'package:flutter/material.dart';
+import 'package:application/utils/translation_en.dart';
+import 'package:application/view/screens/favorites_screen.dart';
+import 'package:application/view/screens/shopping_list_screen.dart';
+
+class HomeScreenViewModel extends ChangeNotifier {
+  final HomeScreenService _homeScreenService = HomeScreenService();
+  String userFirstName = TranslationEN.noData;
+  List<ProductModel> topViewedProducts = [];
+  List<Map<String, dynamic>> menu = [
```

Comment:
A statikusan indexelt listákkal óvatosan. Könnyen nullpointer exception vagy index out of range lehet a használatukból. Statikus indexelés azt értem, hogy a view ahol használod ott úgy hivatkozol rá, hogy menu[0] és menu[1]. Ebben az esetben lehet szerencsésebb lenne, ha ezt a két navigációt ugyan úgy kezelnénk, mint a többit is. Van valami oka annak, hogy ez listába lett téve?